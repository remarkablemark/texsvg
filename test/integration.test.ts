import { promises } from 'fs';
import { exec } from 'child_process';
import { promisify } from 'util';
import { resolve } from 'path';
import texsvg from '../cjs';

const { readFile, unlink } = promises;
const execPromise = promisify(exec);
const bin = resolve(__dirname, '../cjs/bin.js');

describe('texsvg', () => {
  it('can be required using CommonJS or imported using ES Modules', () => {
    expect(texsvg).toBe(texsvg.default);
  });

  it('renders TeX to SVG correctly', async () => {
    const tex = '\\frac{a}{b}';
    expect(await texsvg(tex)).toMatchSnapshot();
  });
});

describe('bin', () => {
  const { argv } = process;
  const processArgv = ['node', 'cjs/bin.js'];

  // mock implementation to override default behavior (prevent log)
  const consoleLog = jest.spyOn(console, 'log').mockImplementation();

  beforeEach(() => {
    consoleLog.mockClear();
  });

  afterAll(() => {
    process.argv = argv;
    consoleLog.mockRestore();
  });

  it('logs SVG to console when 1 argument is passed', async () => {
    // quadratic formula
    const tex = 'x=\\frac{-b\\pm\\sqrt{b^2-4ac}}{2a}';
    process.argv = [...processArgv, tex];
    await require('../cjs/bin');
    expect(consoleLog).toBeCalledWith(await texsvg(tex));
  });

  it('saves SVG to file when 2 arguments are passed ', async () => {
    const tex = 'x';
    const file = 'test.svg';

    await execPromise(`node ${bin} ${tex} ${file}`);
    const svg = await readFile(file, 'utf8');
    await unlink(file);
    expect(svg).toBe(await texsvg(tex));
  });
});
