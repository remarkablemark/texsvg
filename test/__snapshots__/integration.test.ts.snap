// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`texsvg renders TeX to SVG correctly 1`] = `"<svg style="vertical-align:-1.577ex" xmlns="http://www.w3.org/2000/svg" width="2.192ex" height="4.104ex" viewBox="0 -1117 969 1814" xmlns:xlink="http://www.w3.org/1999/xlink"><defs><path id="a" d="M33 157q0 101 76 192t171 92q51 0 90-49 16 30 46 30 13 0 23-8t10-20q0-13-37-160T374 68q0-25 7-33t21-9q9 1 20 9 21 20 41 96 6 20 10 21 2 1 10 1h4q19 0 19-9 0-6-5-27t-20-54-32-50Q436 0 417-8q-8-2-24-2-34 0-57 15t-30 31l-6 15q-1 1-4-1l-4-4q-59-56-120-56-55 0-97 40T33 157m318 171q0 6-5 22t-23 35-46 20q-35 0-67-31t-50-81q-29-79-41-164v-11q0-8-1-12 0-45 18-62t43-18q38 0 75 33t44 51q2 4 27 107t26 111"/><path id="b" d="M73 647q0 10 4 23t12 13q1 0 72 5t73 6q12 0 12-9t-34-143q-8-34-17-70t-15-54l-4-19q0-3 6 3 49 40 101 40 62 0 100-46t39-116q0-111-79-201T173-11q-50 0-91 38T40 150v9q0 21 8 58t49 197q50 197 50 209t-38 14H83q-3 0-7 3t-3 7m263-322v6q0 74-61 74-17 0-35-8t-33-21-26-24-18-22l-6-8-21-86q-22-86-22-122 0-48 24-72 16-16 40-16 33 0 67 32 25 23 40 56t33 105 18 106"/></defs><g stroke="currentColor" fill="currentColor" stroke-width="0"><g transform="scale(1 -1)"><use data-c="1D44E" xlink:href="#a" transform="translate(220 676)"/><use data-c="1D44F" xlink:href="#b" transform="translate(270 -686)"/><path stroke="none" d="M120 220H849V280H120z"/></g></g></svg>"`;
